web/views.py

```diff
@@ -101,3 +183,114 @@ def form_valid(self, form):
 
     def get_success_url(self):
         return reverse('web:studies-list')
+
+class DemographicDataUpdateView(DemographicDataCreateView):
+    """
+    Allows user to update demographic data - but actually creates new version instead of updating old one.
+    """
+    template_name = 'web/demographic-data-update.html'
+
+    def dispatch(self, request, *args, **kwargs):
+        if request.user.is_authenticated():
+            return super().dispatch(request, *args, **kwargs)
+        return HttpResponseRedirect(reverse('login'))
+
+    def get_success_url(self):
+        return reverse('web:demographic-data-update')
+
+    def get_initial(self):
+        """
+        Returns the initial data to use for forms on this view.
+        """
+        demographic_data = self.request.user.latest_demographics or None
+        if demographic_data:
+            demographic_data = demographic_data.__dict__
+            demographic_data.pop('id')
+            demographic_data.pop('uuid')
```

Comment:
We should set the `demographic_data_dict['previous_id']` to `demographic_data.id`.